api/src/services/graphql/index.ts

```diff
@@ -193,23 +199,90 @@ export class GraphQLService {
 	 * Generate the GraphQL schema. Pulls from the schema information generated by the get-schema util.
 	 */
 	async getSchema(): Promise<GraphQLSchema>;
-	async getSchema(type: 'schema'): Promise<GraphQLSchema>;
-	async getSchema(type: 'sdl'): Promise<GraphQLSchema | string>;
-	async getSchema(type: 'schema' | 'sdl' = 'schema'): Promise<GraphQLSchema | string> {
+	async getSchema(type: 'schema', attempt?: number): Promise<GraphQLSchema>;
+	async getSchema(type: 'sdl', attempt?: number): Promise<GraphQLSchema | string>;
+	async getSchema(type: 'schema' | 'sdl' = 'schema', attempt = 0): Promise<GraphQLSchema | string> {
+		const MAX_ATTEMPTS = 3;
```

Comment:
I believe this can be moved to outside the service class next to the other constants, it does not look like it is needed to be defined within the function. It can be renamed to `MAX_SCHEMA_GENERATION_ATTEMPTS` for clarity.

Reply:
The mutex implementation is replicated from https://github.com/directus/directus/blob/436f0a8e53461f2107d4dce55c37a16522c387bd/api/src/utils/get-schema.ts#L34, I think it's fine to leave it within this function as the usage of it is scoped